fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java

```diff
@@ -197,6 +198,8 @@ public static AggregateFunction rectifyAggregationFunction(AggregateFunction fn,
                         ArithmeticExpr.getReturnTypeOfDecimal(ArithmeticExpr.Operator.DIVIDE, (ScalarType) argType,
                                 DECIMAL128P38S0);
                 returnType = triple.returnType;
+            } else if (fn.functionName().equals(FunctionSet.APPROX_TOP_K)) {
+                returnType = new ArrayType(FunctionSet.APPROX_TOP_N_RET_TYPE_BUILDER.apply(argType));
```

Comment:
why not put new ArrayType into apply function?

Reply:
Yes, that's better.